app/src/main/java/org/oppia/android/app/utility/datetime/DateTimeUtil.kt

```diff
@@ -30,4 +31,49 @@ class DateTimeUtil @Inject constructor(
     /** Returns [DateTimeUtil] for the current Dagger graph. */
     fun getDateTimeUtil(): DateTimeUtil
   }
+
+  /**
+   * Returns the readable string of the duration from the provided time in [Long].
+   */
+  fun timeAgoFromTimestamp(timestamp: Long, referenceTime: Long): String {
+    val diff = (currentDate() - timestamp) / MILLI_SECONDS
```

Comment:
For this & the constants below, could we use ``TimeUnit``, instead? That both avoids having to do time math (which can be easily misunderstood and/or written), and may help improve readability. An example of what this might look like:

```kotlin
val diffSeconds = TimeUnit.MILLISECONDS.toSeconds(currentTimeMillis - timestampMillis)
return when {
  diffSeconds < TimeUnit.MINUTES.toSeconds(1) -> ...
  diffSeconds < TimeUnit.HOURS.toSeconds(1) -> ...
  ...
}
```

I think ``TimeUnit`` could also simplify your ``diffTo*`` functions by providing easier conversions. There might even be a way to generalize this slightly, e.g. something like:

```kotlin
private val timestampFormatters by lazy {
  listOf(
    TimestampFormatter(threshold = TimeMoment.ONE_MINUTE, timeUnit = TimeUnit.SECONDS, formatTimestamp = this::formatSecondsTimestamp),
    TimestampFormatter(threshold = TimeMoment.ONE_HOUR, timeUnit = TimeUnit.MINUTES, formatTimestamp = this::formatMinutesTimestamp),
    ...
  )
}

fun formatTimeAgoTimestamp(timestampMillis: Long): String {
  val timestamp = TimeMoment(amount = timestampMillis, unit = TimeUnit.MILLISECONDS)
  val currentTime = TimeMoment(amount = oppiaClock.getCurrentTimeMillis(), unit = TimeUnit.MILLISECONDS)
  val timeSpent = currentTime - timestamp
  return timestampFormatters.firstOrNull {
    it.canFormatWithinThreshold(timeSpent)
  }?.formatString(timeSpent) ?: error("Encountered impossible time: $timeSpent (for time: $timestamp)")
}

private class TimestampFormatter(val threshold: TimeSpan, val timeUnit: TimeUnit, val formatTimestamp: (TimeMoment) -> String) {
  fun canFormatWithinThreshold(timestamp: TimeMoment): Boolean = timestamp.isEarlierThan(threshold)

  fun format(timestamp: TimeMoment): String = formatTimestamp(timestamp.convertTo(timeUnit))
}

private fun formatSecondsTimestamp(seconds: TimeMoment): String =
  resourceHandler.getStringInLocale(R.string.just_now)

private fun formatMinutesTimestamp(minutes: TimeMoment): String {
  return resourceHandler.getQuantityStringInLocaleWithWrapping(
    R.plurals.minutes_ago, minutes.asInt, minutes.asString
  )
}
...

// A simple alternative for Java 8-only time utilities.
private data class TimeMoment(val amount: Long, val unit: TimeUnit) {
  val inMilliseconds: Long
    get() = TimeUnit.MILLISECONDS.convert(amount, unit)

  val asInt: Int
    get() = amount.toInt()

  val asString: String
    get() = amount.toString()

  fun isEarlierThan(other: TimeMoment): Boolean = inMilliseconds < other.inMilliseconds

  fun convertTo(unit: TimeUnit): TimeMoment = TimeMoment(unit.convert(amount, this.unit), unit)

  operator fun minus(other: TimeMoment): TimeMoment =
    TimeMoment(inMilliseconds - other.inMilliseconds, TimeUnit.MILLISECONDS)

  companion object {
    val ONE_MINUTE = TimeMoment(amount = 1, unit = TimeUnit.MINUTES)
    val ONE_HOUR = TimeMoment(amount = 1, unit = TimeUnit.HOURS)
    ...
  }
}
```

(As this is just a concept, more work & reformatting is needed).